scripts/us_bea/states_gdp/import_industry_data_and_gen_mcf.py

```diff
@@ -0,0 +1,163 @@
+        """Downloads ZIP file, extracts the desired CSV, and puts it into a data
+        frame. Stores that data frame in the instance raw_df variable.
+        """
+        # Open zip file from link.
+        resp = urlopen(self._ZIP_LINK)
+
+        # Read the file, interpret it as bytes, and create a ZipFile instance
+        # from it for easy handling.
+        zip_file = zipfile.ZipFile(io.BytesIO(resp.read()))
+
+        # Open the specific desired file (CSV) from the folder, and decode it.
+        # This results in a string representation of the file. Interpret that
+        # as a CSV, and read it into a DF.
+        data = zip_file.open(self._STATE_QUARTERLY_INDUSTRY_GDP_FILE).read()
+        data = data.decode('utf-8')
+        data = list(csv.reader(data.splitlines()))
+        self.raw_df = pd.DataFrame(data[1:], columns=data[0])
+
+    def process_data(self, raw_data=None):
+        """Cleans raw_df and converts it from wide to long format.
+
+        Args:
+            raw_data (optional): raw data frame to be used as starting point
+            for cleaning. If argument is left unspecified, instance self.raw_df
+            is used instead.
+
+        Raises:
+            ValueError: The instance raw_df data frame has not been initialized
+            and no other raw_data was passed as argument. This is probably
+            caused by not having called download_data.
+        """
+        if raw_data is not None:
+            self.raw_df = raw_data
+        if self.raw_df is None:
+            raise ValueError("Uninitialized value of raw data frame. Please "
+                             "check you are calling download_data before "
+                             "process_data.")
+        df = self.raw_df.copy()
+
+        # Filters out columns that are not US states (e.g. New England).
+        df = df[df['GeoName'].isin(self._US_STATES)]
+
+        # Gets columns that represent quarters, e.g. 2015:Q2, by matching
+        # against a regular expression.
+        all_quarters = [q for q in df.columns if re.match(r"\d\d\d\d:Q\d", q)]
+
+        # Convert table from wide to long format.
+        df = pd.melt(df, id_vars=['GeoFIPS', 'IndustryClassification'],
+                     value_vars=all_quarters,
+                     var_name='Quarter')
+
+        df['Quarter'] = df['Quarter'].apply(self._date_to_obs_date)
+        df['GeoId'] = df['GeoFIPS'].apply(self._convert_geoid)
+
+        df = df[df['IndustryClassification'] != "..."]
+
+        df['NAICS'] = df['IndustryClassification'].apply(self._convert_industry_class)
+        df['value'] = df['value'].apply(self._value_converter)
+        df = df[df['value'] >= 0]
+
+        self.clean_df = df.drop(["GeoFIPS", "IndustryClassification"], axis=1)
+
+    @staticmethod
+    def _value_converter(val):
+        """Converts value to float type, and filters out missing values. Missing
+        values are marked by a latter enclosed in parentheses, e.g., "(D)".
```

Comment:
Nitpick: spelling latter -> letter

Does the enclosed letter have a meaning?

Reply:
Yes, for example (D) denotes "Disclosure Avoidance." Added this to the README for clarity.